### ISSUE TITLE
Any chance for an iPhoneSimulator build?

### ISSUE THREAD

**Issue by karalabe** [tpoechtrager/cctools-port#15]
I tried building the cross tools against the iOS simulator SDK. It successfully compiles everything, but then fils in the final step when checking the toolchain.

```
*** checking toolchain ***

ld: file not found: /usr/lib/system/libsystem_kernel.dylib for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Indeed that particular file does not exist, as it's named `libsystem_sim_kernel.dylib` in the simulator SDK. Any ideas where I should look to correct this one last issue?

*Out of curiosity I tried simlinking the above file to the version without `_sim_` in the name, but it got me an infinite linker loop or something `i386-apple-darwin11-ld: ./ld.hpp:109: const ld::File::Ordinal ld::File::Ordinal::nextCounter() const: Assertion `counter() < 0xffff' failed.`*

**Comment by dmigous** [comment]
Did you try to use this project https://github.com/tpoechtrager/osxcross for build for Simulator? I didn't try though, but I think it will compile your code for Simulator

**Comment by tpoechtrager** [comment]
> Out of curiosity I tried simlinking the above file to the version without _sim_ in the name, but it got me an infinite linker loop or something i386-apple-darwin11-ld: ./ld.hpp:109: const ld::File::Ordinal ld::File::Ordinal::nextCounter() const: Assertioncounter() < 0xffff' failed.`

Renaming files may indeed cause an infinite loop, don't do that :-).

> Any ideas where I should look to correct this one last issue?

`Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk/usr/lib/system`.

**Comment by karalabe** [comment]
The simulator SDK is actually missing that file. I am kind of assuming that they meant to link against the live system libs, so if I moved the system `libsystem_kernel.dylib` from /usr/lib into there, it worked properly. I managed to cross compile our quit large stuff and run in in the simulator, however there are a couple issues which are probably rooted in out code, but I can't say for sure until I investigate. I'll keep this thread open and report back if the cross compile indeed works correctly with this workaround or if there are some hidden issues.

**Comment by tpoechtrager** [comment]
The missing file is in the path I posted. You do not need to link against the system libraries (... I think).

**Comment by karalabe** [comment]
Whoah, it ain't there on my machine. Hmm... let me investigate this a bit :D

**Comment by tpoechtrager** [comment]
I am using Xcode_7.1.1.dmg.

**Comment by karalabe** [comment]
Note, the folder and all *is* there, just my file is named `libsystem_sim_kernel.dylib`, not `libsystem_kernel.dylib`, i.e. an extra `_sim_` in there.

**Comment by karalabe** [comment]
Also just asked a friend a moment ago, he also only has the `_sim_` version.

**Comment by tpoechtrager** [comment]
Oh never mind, misread the file name. :-/ Guess linking against the system libraries is the right way then.

**Comment by karalabe** [comment]
Eh, it happens :) Will report back whether all is fine or not :)

**Comment by tpoechtrager** [comment]
You could also take the `libsystem_*` libraries from the Mac OS X SDK (attention: `.tbd` stubs!). Probably a cleaner solution.

**Comment by karalabe** [comment]
I've played with this code a while and I've integrated it into my [Go cross compiler](https://github.com/karalabe/xgo). It seems to behave all right. There have been some issues being reported with some [mach_msg_trap](https://github.com/ethereum/go-ethereum/issues/2069) errors, not sure where the issue lies in that, but I'll open a new bug if it points to the cross compilation package :)

Btw, thanks for your awesome work on these osx/ios cross packages, they really allow be to build very cool and powerful stuff on top of them :) Cheers!
